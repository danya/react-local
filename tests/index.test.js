const babel = require('@babel/core')
const plugin = require('../src')

function transform(input) {
  return babel.transformSync(input, {
    plugins: [plugin()]
  }).code
}

it('creates local variables for named imports', () => {
  const input = `
    import React, {useState, useEffect} from 'react';
  `
  const output = transform(input)
  expect(output).toMatchSnapshot()
})

it('supports import aliasing', () => {
  const input = `
    import React, {useState as foo} from 'react';
  `
  const output = transform(input)
  expect(output).toMatchSnapshot()
})

it('supports import with no default import', () => {
  const input = `
    import {useState} from 'react';
  `
  const output = transform(input)
  expect(output).toMatchSnapshot()
})

it('supports mixed imports', () => {
  const input = `
    import React, {useEffect} from 'react'
    import {useState} from 'react';
  `
  const output = transform(input)
  expect(output).toMatchSnapshot()
})

it('transforms React.createElement calls', () => {
  const input = `
    import React from 'react';
    const element = React.createElement('div');
  `
  const output = transform(input)
  expect(output).toMatchSnapshot()
})
