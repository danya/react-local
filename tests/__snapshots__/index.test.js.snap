// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creates local variables for named imports 1`] = `
"import React from 'react';
const {
  useState,
  useEffect
} = React;"
`;

exports[`supports import aliasing 1`] = `
"import React from 'react';
const {
  useState: foo
} = React;"
`;

exports[`supports import with no default import 1`] = `
"import _React from 'react';
const {
  useState
} = _React;"
`;

exports[`supports mixed imports 1`] = `
"import React from 'react';
const {
  useState
} = React;
const {
  useEffect
} = React;"
`;

exports[`transforms React.createElement calls 1`] = `
"import React from 'react';
const _createElement = React.createElement;

const element = _createElement('div');"
`;
